extensions/widgets/macbutton/macbutton.lcb

```diff
@@ -0,0 +1,214 @@
+    set the join style of this canvas to "bevel"
+    stroke rectangle path of expandRectangle(tBounds, -kBorderWidth / 2) on this canvas
+    variable tPath as Path
+    -- put path kSvgIcon into tPath
+    --constrainPathToRect(expandRectangle(tBounds, -2 * kBorderWidth), tPath)
+    fill tPath on this canvas
+    // Draw the control name
+    put solid paint with color [1, 1, 1, 0.1] into tFillPaint
+    put solid paint with color [0.25, 0.25, 0.25] into tStrokePaint
+    variable tNameString as String
+    if mLabel is empty then
+        put my name into tNameString
+    else
+        put mLabel into tNameString
+    end if
+    set the font of this canvas to my font
+    
+    variable tTextBounds as Rectangle
+    put the image bounds of text tNameString on this canvas into tTextBounds
+    translate this canvas by [ the width of tBounds / 2, the height of tBounds / 2]
+    translate this canvas by [ -(the width of tTextBounds / 2), the height of tTextBounds / 2]
+    set the paint of this canvas to tFillPaint
+    
+    variable tRect as Rectangle
+    put expandRectangle(tTextBounds, kBorderWidth) into tRect
+    fill rounded rectangle path of tRect with radius 5 on this canvas
+    set the paint of this canvas to solid paint with color [0.0, 0.0, 0.0]
+    fill text tNameString at center of tRect on this canvas
+end handler
+
+/**/
+
+public handler OnSave(out rProperties as Array)
+	put mLabel into rProperties["label"]
+end handler
+
+public handler OnLoad(in pProperties as Array)
+	put pProperties["label"] into mLabel
+end handler
+
+/**/
+
+private handler SetLabel(in pLabel as String) returns nothing
+	put pLabel into mLabel
+	if mButtonView is not nothing then
+		unsafe
+			UpdateButtonView()
+		end unsafe
+	end if
+end handler
+
+/****/
+
+private handler ButtonActionCallback(in pSender as ObjcObject, in pContext as optional any) returns nothing
+	post "mouseUp"
+end handler
+
+/****/
+
+private type NSUInteger is CULong
+private foreign handler ObjC_NSButtonAlloc() returns ObjcRetainedId binds to "objc:NSButton.+alloc"
+private foreign handler ObjC_NSButtonInit(in pObj as ObjcId) returns ObjcId binds to "objc:NSButton.-init"
+private foreign handler ObjC_NSButtonSetButtonType(in pObj as ObjcId, in pStyle as NSUInteger) returns nothing binds to "objc:NSButton.-setButtonType:"
+private foreign handler ObjC_NSButtonSetTitle(in pObj as ObjcId, in pTitle as ObjcId) returns nothing binds to "objc:NSButton.-setTitle:"
+private foreign handler ObjC_NSButtonSetTarget(in pObj as ObjcId, in pTarget as ObjcId) returns nothing binds to "objc:NSButton.-setTarget:"
+private foreign handler ObjC_NSButtonSetAction(in pObj as ObjcId, in pAction as UIntPtr) returns nothing binds to "objc:NSButton.-setAction:"
+
+private unsafe handler CreateButtonView()
+	variable tButtonView as ObjcObject
+	put ObjC_NSButtonAlloc() into tButtonView
+	put ObjC_NSButtonInit(tButtonView) into tButtonView
+	ObjC_NSButtonSetButtonType(tButtonView, 7)
```

Comment:
You could maybe put all these button types in as constants and then use the named one, or just say that this 7 comes from https://developer.apple.com/documentation/appkit/nsbutton.buttontype/2097066-momentarypushin